src/main/kotlin/com/makarytskyi/rentcar/repository/CarRepository.kt

```diff
@@ -0,0 +1,26 @@
+package com.makarytskyi.rentcar.repository
+
+import com.makarytskyi.rentcar.model.Car
+import org.springframework.stereotype.Repository
+
+@Repository
+interface CarRepository {
+
+    fun findById(id: String): Car?
+
+    fun save(car: Car): Car
+
+    fun deleteById(id: String)
+
+    fun findAll(): List<Car>
+
+    fun findTheMostExpensiveAvailableCar(): Car?
+
+    fun findTheCheapestAvailableCar(): Car?
+
+    fun findAllByMark(mark: String): List<Car>
```

Comment:
findAllByBrand/Model?

Reply:
Will it be better to separate on 2 methods findAllByBrand and findAllByModel or create findAllByBrandAndModel?

Reply:
findAllByModel sounds strange. Why to seach for Octavia in Audi cars and for A4 in Skoda cars?)
findAllByBrand + findAllByBrandAndModel sounds good to me